rust/processor/src/parquet_processors/move_resource_processor.rs

```diff
@@ -0,0 +1,155 @@
+        if let ParquetStruct::MoveResource(move_resource) = parquet_struct {
+            self.buffer_size_bytes += move_resource.size_of();
+        }
+    }
+    
+    async fn handle_transaction(
+        &mut self,
+        gcs_client: &GCSClient,
+        changes: ParquetData
+    ) -> Result<ParquetProcessingResult> {
+            let last_transaction_timestamp = changes.last_transaction_timestamp;
+            let parquet_structs: Vec<ParquetStruct> = changes.data.clone();
+            let txn_version_to_struct_count = changes.transaction_version_to_struct_count.clone();
+            self.transaction_version_to_struct_count.extend(txn_version_to_struct_count.clone());
+    
+            let schema = self.get_schema(parquet_structs.first().unwrap().clone());
+            let table_name = self.get_file_path_str_from_struct(&parquet_structs.clone().first().unwrap().clone());
+            
+            let file_path = table_name.clone() + ".parquet";
+            let mut file: File = File::options().create(true).write(true).open(&file_path)?;
+    
+            if self.writer.is_none() {
+                let props = WriterProperties::builder()
+                    .set_compression(parquet::basic::Compression::LZ4) 
+                    .build();
+                let props_arc = Arc::new(props.clone());
+                self.writer = Some(SerializedFileWriter::new(file.try_clone()?, schema.clone(), props_arc)?);
+            }
+            let mut writer: SerializedFileWriter<File> = self.writer.take().unwrap(); 
+            info!("Current buffer size:  {:?}", self.buffer_size_bytes);
+            for parquet_struct in parquet_structs.clone() {
+                self.buffer.push(parquet_struct.clone());
+                self.update_buffer_size(&parquet_struct);
+    
+                if self.buffer_size_bytes >= MAX_FILE_SIZE.try_into().unwrap() {  
+                    let starting_version = self.get_version(self.buffer.first().unwrap().clone());
+                    let ending_version = self.get_version(self.buffer.last().unwrap().clone());
+
+                    info!("uploading parquet file with starting version for {} : {} and ending version: {}", 
+                        table_name.clone(),
+                        starting_version,
+                        ending_version
+                    );
+                    let new_file_path: PathBuf = PathBuf::from(format!("{}_{}.parquet", table_name.clone(), chrono::Utc::now().timestamp()));
+                    rename(file_path.clone(), new_file_path.clone())?;
+                    
+                    let mut row_group_writer = writer.next_row_group()?;
+                    let struct_buffer: Vec<_> = self.buffer.iter()
+                        .filter_map(|item| {
+                            if let ParquetStruct::MoveResource(move_resource) = item {
+                                Some(move_resource.clone())
+                            } else {
+                                None
```

Comment:
This is the only different lines of code among all three processors: move_resources, transactions, and write_set_changes. I was trying to move this code to ParquetHandlerTrait but with those fields from this struct, it seems not straightforward. @CapCap any insight on how I can reduce this boilerplate?!